src/main/java/in/org/projecteka/hiu/consent/ConsentService.java

```diff
@@ -51,9 +56,31 @@
     private final ConceptValidator conceptValidator;
     private final GatewayServiceClient gatewayServiceClient;
 
-
+    private Map<ConsentStatus, ConsentTask> consentTasks = new HashMap<>();
     private static final Logger logger = LoggerFactory.getLogger(ConsentService.class);
 
+    public ConsentService(ConsentManagerClient consentManagerClient,
+                          HiuProperties hiuProperties,
+                          ConsentRepository consentRepository,
+                          DataFlowRequestPublisher dataFlowRequestPublisher,
+                          DataFlowDeletePublisher dataFlowDeletePublisher,
+                          PatientService patientService,
+                          CentralRegistry centralRegistry,
+                          HealthInformationPublisher healthInformationPublisher,
+                          ConceptValidator conceptValidator,
+                          in.org.projecteka.hiu.consent.GatewayServiceClient gatewayServiceClient) {
+        this.consentManagerClient = consentManagerClient;
+        this.hiuProperties = hiuProperties;
+        this.consentRepository = consentRepository;
+        this.dataFlowRequestPublisher = dataFlowRequestPublisher;
+        this.dataFlowDeletePublisher = dataFlowDeletePublisher;
+        this.patientService = patientService;
+        this.centralRegistry = centralRegistry;
+        this.healthInformationPublisher = healthInformationPublisher;
+        this.conceptValidator = conceptValidator;
+        this.gatewayServiceClient = gatewayServiceClient;
```

Comment:
Any reason to remove @AllArgsConstructor?

Reply:
because there is an additional map, which I need to set in the postContruct. Lombok does not allow exclusion of fields.

Reply:
Ok. Got it.

Reply:
there is going to be some bit refactoring on the HIU codebase. ConsentService is getting way too big, and all sorts of concerns are getting coded over here. Some of the attributes/constructor arguments will change.